File: Wox.UI.Electron/src/enums/WoxSelectionTypeEnum.ts
import { BaseEnum } from "./base/BaseEnum.ts"

export type WoxSelectionType = string

export class WoxSelectionTypeEnum extends BaseEnum {
  static readonly WoxSelectionTypeText = WoxSelectionTypeEnum.define("text", "text")
  static readonly WoxSelectionTypeFile = WoxSelectionTypeEnum.define("file", "file")
}

File: Wox.UI.Flutter/wox/lib/enums/wox_selection_type_enum.dart
typedef WoxSelectionType = String;

enum WoxSelectionTypeEnum {
  WOX_SELECTION_TYPE_TEXT("text", "text"),
  WOX_SELECTION_TYPE_FILE("file", "file");

  final String code;
  final String value;

  const WoxSelectionTypeEnum(this.code, this.value);

  static String getValue(String code) => WoxSelectionTypeEnum.values.firstWhere((activity) => activity.code == code).value;
}
